### PR TITLE
cleanup(scap,sinsp): move suppressed comm/tids to sinsp

### PR DESCRIPTION
**Pull request by gnosek**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

/area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Suppressed tid/comm support is (was) the only place in scap (except for some testing infrastructure) that cares about the actual event schema and feels bolted on. IMHO it looks much nicer in libsinsp, where we can use an actual set data structure.

**Does this PR introduce a user-facing change?**:

```release-note
BREAKING CHANGE: scap_get_stats() no longer exposes information on suppressed tids/events (sinsp::get_capture_stats() still does)
```

### CODE REVIEW

**Comment by gnosek** [comment]
cc @simonhf since you did the much-debated tid cache :)

File: userspace/libscap/scap.c [hunk 1]
```diff
@@ -385,8 +368,6 @@ int32_t scap_get_stats(scap_t* handle, OUT scap_stats* stats)
 	stats->n_drops_pf = 0;
 	stats->n_drops_bug = 0;
 	stats->n_preemptions = 0;
-	stats->n_suppressed = handle->m_platform->m_suppress.m_num_suppressed_evts;
```
Comment (FedeDP) [inline userspace/libscap/scap.c:388]: This is a breaking change, right? (i am ok with that, just please highlight it in the release notes!)

File: userspace/libsinsp/sinsp_suppress.cpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+{
+	m_suppressed_tids.emplace(tid);
+}
+
+bool libsinsp::sinsp_suppress::check_suppressed_comm(uint64_t tid, const std::string &comm)
+{
+	if(m_suppressed_comms.find(comm) != m_suppressed_comms.end())
+	{
+		m_suppressed_tids.insert(tid);
+		m_num_suppressed_events++;
```
Comment (FedeDP) [inline userspace/libsinsp/sinsp_suppress.cpp:40]: Where is `m_num_suppressed_events` used?

**Review by FedeDP — COMMENTED** [review]
PR LGTM, left a couple of comments!

File: userspace/libscap/scap.c (hunk 1 above)
Comment (gnosek) [reply userspace/libscap/scap.c:388]: Good point. I probably wanted to populate the fields from sinsp and it seemed to slip ;<

Still, even if I restore these fields, they won't be set by raw scap, so it's still technically a breaking change.

File: userspace/libsinsp/sinsp_suppress.cpp (hunk 2 above)
Comment (gnosek) [reply userspace/libsinsp/sinsp_suppress.cpp:40]: Right. Nowhere ;) Will fix.

File: userspace/libscap/scap.c (hunk 1 above)
Comment (FedeDP) [reply userspace/libscap/scap.c:388]: > so it's still technically a breaking change.

Agree! I'd highlight it in the release notes and we are good to go!

File: userspace/libsinsp/sinsp.cpp [hunk 3]
```diff
@@ -1964,6 +1942,8 @@ void sinsp::get_capture_stats(scap_stats* stats) const
 {
 	/* On purpose ignoring failures to not interrupt in case of stats retrieval failure. */
 	scap_get_stats(m_h, stats);
+	stats->n_suppressed = m_suppress.get_num_suppressed_events();
```
Comment (FedeDP) [inline userspace/libsinsp/sinsp.cpp:1945]: I can't say i like it (the fact that `scap_stats` defines a structure where 2 fields are now filled/owned by itself) but i understand this is the only way to achieve this cleanup.
Let's say that, hoping that in the future libscap won't exist anymore, this is a good step in that direction :)

**Review by FedeDP — APPROVED** [review]
/approve

**Review by incertum — APPROVED** [review]
/approve

**Pull request merged**